File: utils/string_utils.c
// utils/string_utils.c

#include <ctype.h>
#include <string.h>

// Capitalizes the first character and lowercases the rest
void capitalize(const char* input, char* output) {
    size_t len = strlen(input);
    if (len == 0) {
        output[0] = '\0';
        return;
    }

    output[0] = toupper(input[0]);
    for (size_t i = 1; i < len; ++i) {
        output[i] = tolower(input[i]);
    }
    output[len] = '\0';
}

File: utils/string_utils.py
# utils/string_utils.py
def capitalize(text):
    """
    Capitalizes the first character of the string and lowers the rest.
    Example: 'hello WORLD' → 'Hello world'
    """
    if not isinstance(text, str):
        raise TypeError("Input must be a string")
    return text[:1].upper() + text[1:].lower()
